diff --git a/node_modules/@strapi/admin/admin/src/content-manager/components/CollectionTypeFormWrapper/index.js b/node_modules/@strapi/admin/admin/src/content-manager/components/CollectionTypeFormWrapper/index.js
index 5c92a9d..47d2689 100644
--- a/node_modules/@strapi/admin/admin/src/content-manager/components/CollectionTypeFormWrapper/index.js
+++ b/node_modules/@strapi/admin/admin/src/content-manager/components/CollectionTypeFormWrapper/index.js
@@ -57,6 +57,11 @@ const CollectionTypeFormWrapper = ({ allLayoutData, children, slug, id, origin }
   const requestURL =
     isCreatingEntry && !origin ? null : `/content-manager/collection-types/${slug}/${origin || id}`;
 
+  const currentContentTypeLayout = get(allLayoutData, ['contentType'], {});
+  const hasVersions = () => {
+    return get(currentContentTypeLayout, ['pluginOptions', 'versions', 'versioned'], false);
+  };
+
   const cleanReceivedData = useCallback((data) => {
     const cleaned = removePasswordFieldsFromData(
       data,
@@ -216,6 +221,7 @@ const CollectionTypeFormWrapper = ({ allLayoutData, children, slug, id, origin }
        */
       const endPoint = isCloning
         ? `/content-manager/collection-types/${slug}/clone/${origin}`
+        : hasVersions ? `/content-versioning/${slug}/save${rawQuery}`
         : `/content-manager/collection-types/${slug}`;
       try {
         // Show a loading button in the EditView/Header.js && lock the app => no navigation
@@ -223,8 +229,12 @@ const CollectionTypeFormWrapper = ({ allLayoutData, children, slug, id, origin }
 
         const { id, ...restBody } = body;
 
+        if (hasVersions) {
+                body.isDuplicatingEntry = isCreatingEntry && !!origin
+              }
+
         const { data } = await post(endPoint, isCloning ? restBody : body, {
-          params: query,
+          params: hasVersions ? rawQuery : query,
         });
 
         trackUsageRef.current('didCreateEntry', trackerProperty);
@@ -321,14 +331,14 @@ const CollectionTypeFormWrapper = ({ allLayoutData, children, slug, id, origin }
 
   const onPut = useCallback(
     async (body, trackerProperty) => {
-      const endPoint = `/content-manager/collection-types/${slug}/${id}`;
+      const endPoint = hasVersions ? `/content-versioning/${slug}/save${rawQuery}` : `/content-manager/collection-types/${slug}/${id}`;
 
       try {
         trackUsageRef.current('willEditEntry', trackerProperty);
 
         dispatch(setStatus('submit-pending'));
 
-        const { data } = await put(endPoint, body);
+        const { data } = hasVersions ? await post(endPoint, body) : await put(endPoint, body);
 
         trackUsageRef.current('didEditEntry', { trackerProperty });
         toggleNotification({
@@ -343,6 +353,10 @@ const CollectionTypeFormWrapper = ({ allLayoutData, children, slug, id, origin }
 
         dispatch(setStatus('resolved'));
 
+        if(hasVersions){
+          replace(`/content-manager/collectionType/${slug}/${data.id}${rawQuery}`)
+        }
+
         return Promise.resolve(data);
       } catch (err) {
         trackUsageRef.current('didNotEditEntry', { error: err, trackerProperty });
diff --git a/node_modules/@strapi/admin/admin/src/content-manager/components/EditViewDataManagerProvider/index.js b/node_modules/@strapi/admin/admin/src/content-manager/components/EditViewDataManagerProvider/index.js
index 8b3d9b8..0a25b84 100644
--- a/node_modules/@strapi/admin/admin/src/content-manager/components/EditViewDataManagerProvider/index.js
+++ b/node_modules/@strapi/admin/admin/src/content-manager/components/EditViewDataManagerProvider/index.js
@@ -81,6 +81,10 @@ const EditViewDataManagerProvider = ({
     return get(currentContentTypeLayout, ['options', 'draftAndPublish'], false);
   }, [currentContentTypeLayout]);
 
+  const hasVersions = useMemo(() => {
+    return get(currentContentTypeLayout, ['pluginOptions', 'versions', 'versioned'], false);
+  }, [currentContentTypeLayout]);
+
   const shouldNotRunValidations = useMemo(() => {
     return hasDraftAndPublish && !initialData.publishedAt;
   }, [hasDraftAndPublish, initialData.publishedAt]);
@@ -663,7 +667,7 @@ const EditViewDataManagerProvider = ({
         <>
           {!isSaving ? (
             <Prompt
-              when={!isEqual(modifiedData, initialData)}
+              when={!hasVersions && !isEqual(modifiedData, initialData)}
               message={formatMessage({ id: 'global.prompt.unsaved' })}
             />
           ) : null}
